packages/ui-popover/src/Popover/props.ts

```diff
@@ -33,270 +33,300 @@ import type { Shadow, Stacking } from '@instructure/emotion'
-  constrain?: PositionConstraint
-  mountNode?: PositionMountNode
-  positionTarget?: PositionMountNode
-  insertAt?: 'bottom' | 'top'
-  liveRegion?: LiveRegion
-  id?: string
-  shouldAlignArrow?: boolean
-  shouldTrackPosition?: boolean
-  shouldRenderOffscreen?: boolean
-  shouldContainFocus?: boolean
-  shouldReturnFocus?: boolean
-  shouldCloseOnDocumentClick?: boolean
-  shouldCloseOnEscape?: boolean
-  shouldFocusContentOnTriggerBlur?: boolean
-  onShowContent?: (...args: any[]) => any
-  onHideContent?: (...args: any[]) => any
-  onPositioned?: (...args: any[]) => any
-  onPositionChanged?: (...args: any[]) => any
-  onClick?: (...args: any[]) => any
-  onFocus?: (...args: any[]) => any
-  onBlur?: (...args: any[]) => any
-  onKeyDown?: (...args: any[]) => any
-  onKeyUp?: (...args: any[]) => any
-  onMouseOver?: (...args: any[]) => any
-  onMouseOut?: (...args: any[]) => any
-  renderTrigger?: React.ReactNode | ((...args: any[]) => any)
-  children?: React.ReactNode | ((...args: any[]) => React.ReactNode)
-  elementRef?: (element: Element | null) => void
-}
-
-type PopoverProps = PopoverOwnProps & BidirectionalProps
-
-type PropKeys = keyof PopoverOwnProps
-
-type AllowedPropKeys = Readonly<Array<PropKeys>>
-
-const propTypes: PropValidators<PropKeys> = {
   /**
    * Whether or not the `<Popover />` content is shown
    */
-  isShowingContent: PropTypes.bool,
+  isShowingContent?: boolean
+
   /**
    * Whether or not to show the content by default, when uncontrolled
    */
-  defaultIsShowingContent: PropTypes.bool,
+  defaultIsShowingContent?: boolean
+
   /**
    * The action that causes the content to display (`click`, `hover`, `focus`)
    */
-  on: PropTypes.oneOfType([
-    PropTypes.oneOf(['click', 'hover', 'focus']),
-    PropTypes.arrayOf(PropTypes.oneOf(['click', 'hover', 'focus']))
-  ]),
+  on?: ('click' | 'hover' | 'focus') | ('click' | 'hover' | 'focus')[]
+
   /**
    * Whether or not an arrow pointing to the trigger should be rendered
    */
-  withArrow: PropTypes.bool,
+  withArrow?: boolean
+
   /**
    * Color variant of the popover content
    */
-  color: PropTypes.oneOf(['primary', 'primary-inverse']),
-  /**
-   * The placement of the content in relation to the trigger
-   */
-  placement: PropTypes.oneOf(placementPropValues),
+  color?: 'primary' | 'primary-inverse'
+
   /**
    * Controls the shadow depth for the `<Popover />`
    */
-  shadow: ThemeablePropTypes.shadow,
+  shadow?: Shadow
+
   /**
    * Controls the z-index depth for the `<Popover />` content
    */
-  stacking: ThemeablePropTypes.stacking,
+  stacking?: Stacking
+
   /**
    * A function that returns a reference to the content element
    */
-  contentRef: PropTypes.func,
+  contentRef?: (contentElement: Element | null) => void
+
   /**
    * An element or a function returning an element to focus by default
    */
-  defaultFocusElement: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
+  defaultFocusElement?: UIElement
```

Comment:
I noticed sth strange here: TS can read the types imported from shared-types well and they appear in the Props object correctly, but any type imported from other (component) packages does not. It might be worth exploring.